server/routes/auth-routes.js

```diff
@@ -0,0 +1,85 @@
+const router = require('express').Router();
+const passport = require('passport');
+let client_add = "http://localhost:3000";
+
+// function ensureAuthenticated
+// ensures the user is logged in before it grants access to the api.
+// input: req, the request; res, the response obj; next, the function to be called next
+function ensureAuthenticated(req, res, next)
+{
+    if (req.isAuthenticated()) {
+        return next();
+    } else {
+        res.status(403).json( {"failure": "you are not logged in"});
+    }
+}
+
+// auth login
+router.get('/login', (req,res) => {
+  res.send('Success');
+});
+
+// auth logout
+router.get('/logout', ensureAuthenticated, (req,res) => {
+    req.logout();
+    // TODO: detect whether logout is successful
+    res.send('Success');
+  // handle with passport
+});
+
+// auth with our own Login
+router.get('/local', (req, res) => {
+    res.sent("Error: GET method not supported.");
+});
+
+router.post('/local', passport.authenticate('local'), (req, res) => {
+    res.json({
+        "status": "success",
+        "user": {
+            "username": req.user.username,
+            "email": req.user.email,
+            "firstName": req.user.firstName,
+            "lastName": req.user.lastName
+        }
+    });
+});
+
+// auth with our own signup
+router.get('/signup', (req, res) => {
+  res.render('signup');
+});
+
+// auth with google
+router.get('/google', passport.authenticate('google', {
+  scope: ['profile', 'email']
+}));
+
+// callback route for google to redirect to
+router.get('/google/redirect',
+  passport.authenticate('google', {"failureRedirect": "/google"}),
+  (req,res) => {
+      var name = req.user.firstName + " " + req.user.lastName;
+      res.send("<html><p>you reached the callback URI. </p><button onclick='window.close()'>Close Window</button>" +
+          "<script>window.onload = () => {console.log(\"sending msg\");" +
+          "window.opener.postMessage(\"" + name + "\", \"" + client_add + "\");window.close()};</script></html>");
+  }
+);
+
+router.get('/loginstatus', (req, res) => {
```

Comment:
use case for this route?

Reply:
to get info for the user currently logged in